lib/node_modules/@stdlib/ndarray/base/fill/benchmark/benchmark.2d_blocked_columnmajor.js

```diff
@@ -0,0 +1,139 @@
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+'use strict';
+
+// MODULES //
+
+var bench = require( '@stdlib/bench' );
+var discreteUniform = require( '@stdlib/random/base/discrete-uniform' ).factory;
+var isnan = require( '@stdlib/math/base/assert/is-nan' );
+var pow = require( '@stdlib/math/base/special/pow' );
+var floor = require( '@stdlib/math/base/special/floor' );
+var sqrt = require( '@stdlib/math/base/special/sqrt' );
+var filledarrayBy = require( '@stdlib/array/filled-by' );
+var pkg = require( './../package.json' ).name;
+var fill = require( './../lib' );
+
+
+// VARIABLES //
+
+var types = [ 'float64' ];
+var order = 'column-major';
+var value = 10.0;
+
+
+// FUNCTIONS //
+
+/**
+* Creates a benchmark function.
+*
+* @private
+* @param {PositiveInteger} len - ndarray length
+* @param {NonNegativeIntegerArray} shape - ndarray shape
+* @param {IntegerArray} strides - ndarray strides
+* @param {string} xtype - input ndarray data type
+* @returns {Function} benchmark function
+*/
+function createBenchmark( len, shape, strides, xtype ) {
+	var x;
+
+	x = filledarrayBy( len, xtype, discreteUniform( -100, 100 ) );
+	x = {
+		'dtype': xtype,
+		'data': x,
+		'shape': shape,
+		'strides': strides,
+		'offset': 0,
+		'order': order
+	};
+	return benchmark;
+
+	/**
+	* Benchmark function.
+	*
+	* @private
+	* @param {Benchmark} b - benchmark instance
+	*/
+	function benchmark( b ) {
+		var i;
+
+		b.tic();
+		for ( i = 0; i < b.iterations; i++ ) {
+			fill( x, value );
+			if ( isnan( x.data[ i%len ] ) ) {
+				b.fail( 'should not return NaN' );
+			}
+		}
+		b.toc();
+		if ( isnan( x.data[ i%len ] ) ) {
+			b.fail( 'should not return NaN' );
+		}
+		b.pass( 'benchmark finished' );
+		b.end();
+	}
+}
+
+
+// MAIN //
+
+/**
+* Main execution sequence.
+*
+* @private
+*/
+function main() {
+	var len;
+	var min;
+	var max;
+	var sh;
+	var st;
+	var t1;
+	var f;
+	var i;
+	var j;
+
+	min = 1; // 10^min
+	max = 6; // 10^max
+
+	for ( j = 0; j < types.length; j++ ) {
+		t1 = types[ j ];
+		for ( i = min; i <= max; i++ ) {
+			len = pow( 10, i );
+
+			sh = [ len/2, 2 ];
+			st = [ -1, len/2 ];
+			f = createBenchmark( len, sh, st, t1 );
+			bench( pkg+'::blocked:ndims='+sh.length+',len='+len+',shape=['+sh.join(',')+'],xorder='+order+',xtype='+t1, f );
+
+			sh = [ 2, len/2 ];
+			st = [ -1, 2 ];
+			f = createBenchmark( len, sh, st, t1 );
+			bench( pkg+'::blocked:ndims='+sh.length+',len='+len+',shape=['+sh.join(',')+'],xorder='+order+',xtype='+t1, f );
+
+			len = floor( sqrt( len ) );
+			sh = [ len, len ];
+			st = [ -1, len ];
+			len *= len;
+			f = createBenchmark( len, sh, st, t1 );
+			bench( pkg+'::blocked:ndims='+sh.length+',len='+len+',shape=['+sh.join(',')+'],xorder='+order+',xtype='+t1, f );
```

Comment:
I'd like your thoughts on this, I added strides for blocked benchmarks since we don't have separate kernels applied like assign.

Reply:
Thanks for checking. I don't think we need to go overboard with benchmarks for `fill`. Because it is based on `assign`, should be more or less the same. In which case, how about we just do 2d row/column-major and 2d row/column-major with accessors.

Reply:
Should we ever need more benchmarks, we can add them later, but, for now, we just need some sanity checks. Otherwise, we start needing to make assumptions about the implementation of `assign`, which doesn't feel right to me for this package.